tests/test_bonded.py

```diff
@@ -62,34 +66,30 @@ def test_centroid_restraint(self):
                 precision=precision
             )
 
+    def test_harmonic_bond(self, n_particles=64, n_bonds=35, dim=3):
+        """Randomly connect pairs of particles, then validate the resulting HarmonicBond force"""
+        np.random.seed(125)  # TODO: where should this seed be set?
```

Comment:
Seeds are a difficult topic, setting them to a fixed number has the advantage of making it easy to debug (eg. when tests fail) as it makes everything reproducible, but has the downside of making the method not truly random. Maybe a hybrid approach where we roll a "global" seed (eg. 2020) once a year is the way to go. The annoyances of false positives in testing (failed test that failed because of a random seed in a different part of the code) can't be understated.

Reply:
Hmm, I think we would want to (1) set a random seed so that the execution of the entire test program is deterministic, (2) reduce any sensitivity to the choice of random seed.

To achieve (2), I think we should be running multiple randomized tests rather than a single randomized test, if cheap enough. (The unit tests here are cheap enough to run many times, but some of the integration tests are not.)

To achieve (1), setting the random seed once at the top of the file is sufficient, but may not be convenient (e.g. if you want to reproduce a specific test failure in a file containing many tests). To make it easier to reproduce behavior of specific tests that depend on random state, we can modify the test function signature to accept a seed or random state as an argument.

Reply:
How does pytest work with test_xxx functions that have different default kwargs? Are you able to pass in different values into pytest? (Or is the intention for this to be called from another test_ function? since pytest will automatically run any functions that begin with test_)

Reply:
Yep! PyTest provides a nice way to "parametrize" a test, as in this example https://docs.pytest.org/en/stable/parametrize.html

```python
import pytest


@pytest.mark.parametrize("test_input,expected", [("3+5", 8), ("2+4", 6), ("6*9", 42)])
def test_eval(test_input, expected):
    assert eval(test_input) == expected
```

Initially, my intention would be to do something similar to this example, but using a vector of generated random seeds rather than a vector of hand-written inputs. (That way each individual test "knows" what random state it used, and the whole collection of generated tests should run deterministically.) It may be a good idea for me to take a quick look for other examples of modern pythonic randomized tests.

Reply:
Okay that makes sense, feel free to re-request a review when you think its ready for a second pass (I'll probably just approve it without comments if the changes are mainly clerical)

Reply:
Thanks! I think the tricky issue of test randomization would better addressed in a dedicated subsequent PR -- wasn't intending to make substantive changes to the testing logic in the present PR.